web/modules/custom/hoeringsportal_citizen_proposal/src/Form/ProposalFormAdd.php

```diff
@@ -108,11 +126,27 @@ public function buildProposalForm(array $form, FormStateInterface $formState): a
     return $form;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state): void {
+    if (strlen($form_state->getValue('title')) > $this->getMaxLength('characters_title')) {
+      $form_state->setErrorByName('title', $this->t('Too many characters used.'));
+    }
+
+    if (strlen($form_state->getValue('proposal')) > $this->getMaxLength('characters_proposal')) {
+      $form_state->setErrorByName('proposal', $this->t('Too many characters used.'));
+    }
+
+    if (strlen($form_state->getValue('remarks')) > $this->getMaxLength('characters_remarks')) {
+      $form_state->setErrorByName('remarks', $this->t('Too many characters used.'));
```

Comment:
Less generic error messages may improve the user experience, e.g.

```
The proposal title can contain at most @number character.
```

and similar for the other fields.

In the admin we have a HTML text format on “proposal” and “remarks”. If using HTML for the content (which the citizens cannot do right now) we need the check the length of the actual text content and not just the `strlen` of the content.

Reply:
The error message is posistioned below the field it relates to, so there shouldn't be a problem identifying the problematic field as it is now.

Changing the field to use html/ckeditor will indeed require som testing and tweaks both in the validation handler and the maxlength module.